tests/MongoDB.Driver.Core.Tests/Core/Configuration/ConnectionSettingsTests.cs

```diff
@@ -104,13 +111,16 @@ public void constructor_with_applicationName_should_initialize_instance()
         public void constructor_with_authenticators_should_initialize_instance()
         {
 #pragma warning disable 618
-            var authenticators = new[] { new MongoDBCRAuthenticator(new UsernamePasswordCredential("source", "username", "password")) };
+            var originalAuthenticator = new MongoDBCRAuthenticator(new UsernamePasswordCredential("source", "username", "password"));
+            var authenticatorFactories = new[] { new AuthenticatorFactory(() => originalAuthenticator) };
 #pragma warning restore 618
 
-            var subject = new ConnectionSettings(authenticators: authenticators);
+            var subject = new ConnectionSettings(authenticatorFactories: authenticatorFactories);
 
             subject.ApplicationName.Should().BeNull();
-            subject.Authenticators.Should().Equal(authenticators);
+            subject.AuthenticatorFactories.Should().NotBeNull();
+            var authenticators = subject.AuthenticatorFactories.Select(a => a.Create());
+            authenticators.Single().Should().BeSameAs(originalAuthenticator);
```

Comment:
Replace 121-123 with:

            subject.AuthenticatorFactories.Should().Equal(authenticatorFactories);